src/server/editorServices.ts

```diff
@@ -2095,9 +2117,19 @@ namespace ts.server {
 
             this.printProjects();
 
+            this.telemetryOnOpenFile(info);
             return { configFileName, configFileErrors };
         }
 
+        private telemetryOnOpenFile(scriptInfo: ScriptInfo): void {
+            if (!this.eventHandler || !scriptInfo.isJavaScript() || !addToSeen(this.allJsFilesForOpenFileTelemetry, scriptInfo.path)) {
+                return;
+            }
+
+            const info: OpenFileInfo = { checkJs: !!scriptInfo.getDefaultProject().getSourceFile(scriptInfo.path).checkJsDirective };
+            this.eventHandler({ eventName: OpenFileInfoTelemetryEvent, data: { info } });

```

Comment:
dont you need to send the open file name so that at vscode side they can create a map of seen files and get correct count instead of count = 3 when opening same file 3 times?

Reply:
If we open the same file 3 times we should still only send one event, thanks to `this.allJsFilesForOpenFileTelemetry`.